prov/gni/include/gnix_vector.h

```diff
@@ -201,7 +209,19 @@ int _gnix_vec_resize(gnix_vector_t *vec, uint64_t size);
  * @return -FI_EINVAL	Upon passing a NULL or dead vector
  * @return -FI_ECANCLED Upon attempting to get an empty element
  */
-int _gnix_vec_at(gnix_vector_t *vec, void **element, gnix_vec_index_t index);
+static inline int _gnix_vec_at(gnix_vector_t *vec, void **element,
+			       gnix_vec_index_t index)
+{
+	GNIX_TRACE(FI_LOG_EP_CTRL, "\n");
+
+	if (unlikely(!vec || !element)) {
+		GNIX_WARN(FI_LOG_EP_CTRL, "Invalid parameter to "
+			  "_gnix_vec_at\n");
+		return -FI_EINVAL;
+	} else {
+		return vec->ops->at(vec, element, index);
+	}
```

Comment:
> Consider assigning the ops table rather than using the ops table reference. This could decrease dereference time.

I'll wait to do this until there is evidence that the vector code is a bottle neck.